extensions/wikia/CommunityPage/scripts/firstEditModal.js

```diff
@@ -0,0 +1,130 @@
+require([
+	'jquery',
+	'wikia.ui.factory',
+	'wikia.mustache',
+	'communitypage.templates.mustache',
+	'wikia.nirvana',
+	'wikia.tracker',
+	'wikia.window',
+	'wikia.cookies',
+], function ($, uiFactory, mustache, templates, nirvana, tracker, window, cookies) {
+	'use strict';
+
+	 var track = tracker.buildTrackingFunction({
+		 action: tracker.ACTIONS.CLICK,
+		 category: 'community-page-first-edit',
+		 trackingMethod: 'analytics'
+	 });
+
+	function getUiModalInstance() {
+		var $deferred = $.Deferred();
+
+		uiFactory.init(['modal']).then(function (uiModal) {
+			$deferred.resolve(uiModal);
+		});
+
+		return $deferred;
```

Comment:
Since `uiFactory.int` returns a promise, couldn't you just `return uiFactory.init(['modal']);` here rather than creating another promise object?